###############################################################################
# Copyright 2008 Sun Microsystems, Inc.  All rights reserved.
# Use is subject to license terms.
#
# Permission is hereby granted, free of charge, to any person obtaining a
# copy of this software and associated documentation files (the "Software"),
# to deal in the Software without restriction, including without limitation
# the rights to use, copy, modify, merge, publish, distribute, sublicense,
# and/or sell copies of the Software, and to permit persons to whom the
# Software is furnished to do so, subject to the following conditions:
#
# The above copyright notice and this permission notice (including the next
# paragraph) shall be included in all copies or substantial portions of the
# Software.
#
# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL
# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
# DEALINGS IN THE SOFTWARE.
#

1261271 /usr/dist/exe/gnuemacs doesn't work on s297.

diff -urp -x '*~' -x '*.orig' src/XlibInt.c src/XlibInt.c
--- src/XlibInt.c	2009-03-24 23:25:37.548747000 -0700
+++ src/XlibInt.c	2009-03-24 23:25:49.514829000 -0700
@@ -54,6 +54,12 @@
 #include <direct.h>
 #endif

+/* Xtransint.h is needed for access to an xtrans internal structure.
+ * See the comments in _XFlush for information. This is required for
+ * compatibility between X11R6-xtrans and gnuemacs v18.58.
+ */
+#include <X11/Xtrans/Xtransint.h>
+
 #ifdef XTHREADS
 #include "locking.h"

@@ -653,6 +659,18 @@
	register char *bufindex;
	_XExtension *ext;

+	/* Begin gnuemacs v18.58 compatability fix.
+	 * In x11trans.c, line 2207, in gnuemacs source, the dpy->fd is
+	 * modified.  This was of no consequence in X11R5, but in X11R6,
+	 * the xtrans layer has a copy which must be updated here.
+	 * The reason to perform the update here is because this code
+	 * is executed right after gnuemacs changes the fd variable.
+	 */
+	if (dpy->fd != dpy->trans_conn->fd) {
+		dpy->trans_conn->fd = dpy->fd;
+	}
+	/* End gnuemacs v18.58 compatability fix. */
+
 	/* This fix resets the bufptr to the front of the buffer so
	 * additional appends to the bufptr will not corrupt memory. Since
	 * the server is down, these appends are no-op's anyway but
